WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsSearchingViewController.swift

```diff
@@ -26,29 +26,28 @@ final class CardReaderSettingsSearchingViewController: UIHostingController<CardR
         )
     }()
 
-    required init?(coder: NSCoder) {
-        super.init(coder: coder, rootView: CardReaderSettingsSearchingView())
+    init?(viewModel: PaymentSettingsFlowPresentedViewModel) {
+        guard let viewModel = viewModel as? CardReaderSettingsSearchingViewModel else {
+            return nil
+        }
```

Comment:
This is a failable initialiser, so in its own context, silently returning `nil` is valid behaviour. I agree though, logging if it happens in our flows would still be useful, but perhaps more useful at the call site rather than in the initialiser. That way we have more relevant context in the log.

Any new callers can deal with the potential for nil being returned as is appropriate to them.

All that said, problems like this are most likely to be caught at development time, with the way this is set up...

Added logs in d20e1e6

<img width="881" alt="Screenshot of log in the console" src="https://user-images.githubusercontent.com/2472348/222444096-2f9f23e6-8097-4377-83b2-73b8fe2f866f.png">